pkg/model/application_live_state.go

```diff
@@ -144,3 +144,27 @@ func (s *ApplicationLiveStateSnapshot) determineLambdaAppHealthStatus() {
 	}
 	s.HealthStatus = ApplicationLiveStateSnapshot_HEALTHY
 }
+
+// DetermineApplicationHealthStatus updates the health status of the application based on the health status of its resources.
+func (s *ApplicationLiveStateSnapshot) DetermineApplicationHealthStatus() {
+	app := s.ApplicationLiveState
+	if app == nil {
+		return
+	}
+
+	for _, r := range app.Resources {
+		if r.HealthStatus == ResourceState_UNHEALTHY {
+			s.HealthStatus = ApplicationLiveStateSnapshot_UNHEALTHY
+			return
+		}
+	}
+
+	for _, r := range app.Resources {
+		if r.HealthStatus == ResourceState_UNKNOWN {
+			s.HealthStatus = ApplicationLiveStateSnapshot_UNKNOWN
+			return
+		}
+	}
```

Comment:
why not just merge these 2 for loops into one? 🤔

Reply:
```suggestion
	for _, r := range app.Resources {
		if r.HealthStatus == ResourceState_UNKNOWN {
			s.HealthStatus = ApplicationLiveStateSnapshot_UNKNOWN
			return
		}
		if r.HealthStatus == ResourceState_UNHEALTHY {
			s.HealthStatus = ApplicationLiveStateSnapshot_UNHEALTHY
			return
		}
	}
```

Reply:
@khanhtc1202 

See this:

https://github.com/pipe-cd/pipecd/pull/5454#discussion_r1897664342